issuance/cert.go

```diff
@@ -72,6 +93,78 @@ type ProfileConfig struct {
 	Policies []PolicyConfig `validate:"-"`
 }
 
+// ProfileConfigNew describes the certificate issuance constraints for all issuers.
+//
+// See ProfileConfig for why this is called "New".
+//
+// This struct gets hashed in the CA to allow matching up precert and final cert
+// issuance by the exact profile config. We compute the hash over an ASN.1 encoding
+// because ASN.1 encoding has a canonical form and can omit optional fields (which
+// allows for gracefully adding new fields without changing the hash of existing
+// profile configs). This struct does not get embedded into any certs, CRLs, or
+// other objects, and does not get signed; it's only used internally.
+type ProfileConfigNew struct {
+	// AllowMustStaple, when false, causes all IssuanceRequests which specify the
+	// OCSP Must Staple extension to be rejected.
+	AllowMustStaple bool `asn1:"tag:1,optional"`
+
+	// OmitCommonName causes the CN field to be excluded from the resulting
+	// certificate, regardless of its inclusion in the IssuanceRequest.
+	OmitCommonName bool `asn1:"tag:2,optional"`
+	// OmitKeyEncipherment causes the keyEncipherment bit to be omitted from the
+	// Key Usage field of all certificates (instead of only from ECDSA certs).
+	OmitKeyEncipherment bool `asn1:"tag:3,optional"`
+	// OmitClientAuth causes the id-kp-clientAuth OID (TLS Client Authentication)
+	// to be omitted from the EKU extension.
+	OmitClientAuth bool `asn1:"tag:4,optional"`
+	// OmitSKID causes the Subject Key Identifier extension to be omitted.
+	OmitSKID bool `asn1:"tag:5,optional"`
+
+	MaxValidityPeriod   config.Duration `asn1:"tag:6,optional"`
+	MaxValidityBackdate config.Duration `asn1:"tag:7,optional"`
+
+	IncludeCRLDistributionPoints bool `asn1:"tag:8,optional"`
```

Comment:
hyper-nit: I'd put this closer to the `Omit...` fields, since it concerns the inclusion of a specific extension, rather than having it separated by the validity controls fields.

Also, it needs a comment.